src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -126,9 +128,11 @@
  */
 public final class ElytronXmlParser {
 
-    private static final Supplier<Provider[]> ELYTRON_PROVIDER_SUPPLIER = WildFlySecurityManager.isChecking() ?
-            AccessController.doPrivileged((PrivilegedAction<ServiceLoaderSupplier<Provider>>) () -> new ServiceLoaderSupplier<>(Provider.class, ElytronXmlParser.class.getClassLoader())) :
-            new ServiceLoaderSupplier<>(Provider.class, ElytronXmlParser.class.getClassLoader());
+    private static final Supplier<Provider[]> ELYTRON_PROVIDER_SUPPLIER = ProviderUtil.aggregate(
+            () -> new Provider[] { new WildFlyElytronProvider() },
+            WildFlySecurityManager.isChecking() ?
```

Comment:
I don't think this needs to be privileged since the provider supplier doesn't seem to capture the ACC on construction, unless it does and I just can't see it in this diff...

Reply:
This was switched over to being privileged in ELY-1349: https://github.com/wildfly-security/wildfly-elytron/pull/972

Reply:
The parent class, ```ServiceLoaderSupplier```, captures the ACC on construction.

Reply:
OK that makes sense.